### DISCUSSION TITLE
Bugs

### DISCUSSION THREAD

**Discussion by kvirtu** [givanz/Vvveb#15] [General]
I will post bugs here.
1. If "Page>Add new page" then Link: http://localsite/admin/?module=content/post&type=page
with error on page: _Warning: Undefined array key "template" in /virt/homes/vpu2/htdocs/admin/controller/content/post.php on line 122_ with "Post" "Post list" "Save post"
2. can't change values px in Style
3. /admin/?module=content/posts&type=page . Menu "Page" , but  displayed "posts" and "add post"
4. In "Edit website" how to delete a page ?
5.  In "Edit website" , new pages are not  edited and not saved
6.  menu "tools" link by default /admin/?module=tools/**systeminfo**
7. In menu "tools>cache" buttons "Clear old/all cache" link: /admin/?module=product/product_
8. menu "Post>categories" - titles not showing
9. "Media library" - not saved images. _<br /> <b>Warning</b>: move_uploaded_file(/virt/homes/_
10. "Media library" -  how to create a new folder ? sorting by date ? if it is possible ?
11. Menu "post>Tags" error: Warning: Undefined array key "languages" in /virt/homes/vpu2/htdocs/admin/controller/content/tags.php on line 43
12. Menu "Themes" button "Install Theme" - error:
Fatal error: Uncaught ArgumentCountError: Too few arguments to function Vvveb\System\Cache::get(), 1 passed in /virt/homes/vpu2/htdocs/system/extensions/themes.php on line 95 and exactly 2 expected in /virt/homes/vpu2/htdocs/system/cache.php:56 Stack trace: #0 /virt/homes/vpu2/htdocs/system/extensions/themes.php(95): Vvveb\System\Cache->get('bdb43498f487b3c...') #1 /virt/homes/vpu2/htdocs/admin/controller/theme/market.php(39): Vvveb\System\Extensions\Themes::getMarketList(Array) #2 [internal function]: Vvveb\Controller\Theme\Market->index() #3 /virt/homes/vpu2/htdocs/system/core/frontcontroller.php(190): call_user_func(Array) #4 /virt/homes/vpu2/htdocs/system/core/frontcontroller.php(255): Vvveb\System\Core\FrontController::call('Vvveb\\Controlle...', 'index', '/virt/homes/vpu...') #5 /virt/homes/vpu2/htdocs/system/core/frontcontroller.php(307): Vvveb\System\Core\FrontController::redirect('Theme/Market', 'index') #6 /virt/homes/vpu2/htdocs/system/core/startup.php(321): Vvveb\System\Core\FrontController::dispatch() #7 /virt/homes/vpu2/htdocs/index.php(122): Vvveb\System\Core\start() #8 /virt/homes/vpu2/htdocs/admin/index.php(31): include('/virt/homes/vpu...') #9 {main} thrown in /virt/homes/vpu2/htdocs/system/cache.php on line 56

**Comment by givanz** [comment]
Thanks for the bug report list and feedback.

> If "Page>Add new page" then Link: http://localsite/admin/?module=content/post&type=page with error on page: Warning: Undefined array key "template" in /virt/homes/vpu2/htdocs/admin/controller/content/post.php on line 122 with "Post" "Post list" "Save post"

This warning was caused by a recent change, I fixed it.

> can't change values px in Style

You mean Style tab in page editor? I tried changing font size for an element in px and it worked, can you provide more details on exact input?

> /admin/?module=content/posts&type=page . Menu "Page" , but displayed "posts" and "add post"

Fixed it.

> In "Edit website" how to delete a page ?
> In "Edit website" , new pages are not edited and not saved

This was inherited from VvvebJs and not changed yet but for Vvveb cms I think is better to name it template not to be confused with pages because they are in fact html templates from the installed theme that are used to display different sections from the site including posts/pages/products/cart etc.

I don't think is a good ideea to have delete option in the file manager for theme html pages as deleting a template like content/product.html will break the product page, someone who knows what is doing can delete a html file using the theme code editor or maybe add delete only for templates that are not used as global templates like homepage/post/product etc to avoid breaking the site.

New template is not implemented yet but will be available soon, it will simply allow to duplicate existing template pages or start with the "blank.html" provided by the theme.


> menu "tools" link by default /admin/?module=tools/systeminfo

All parent menu items that have submenus also link directly to a page as a shortcut, not sure what shortcut is best for tools, probably cache will be the most used. 


> In menu "tools>cache" buttons "Clear old/all cache" link: /admin/?module=product/product_

Clear cache is not implemented yet, there is only some automatic clear cache when a new post/page/product is added, this is work in progress.

> menu "Post>categories" - titles not showing

I checked /admin/?module=content/categories&type=post and titles are showing, can you provide more details regarding this issue?


> "Media library" - not saved images. Warning: move_uploaded_file(/virt/homes/

Can you provide more details like complete error message? I thinks recursive write permission is missing for public/media folder


> "Media library" - how to create a new folder ? sorting by date ? if it is possible ?

Media library currently does not have many features implemented like new folder and sorting, these will be added in future releases.


> Menu "post>Tags" error: Warning: Undefined array key "languages" in /virt/homes/vpu2/htdocs/admin/controller/content/tags.php on line 43

I fixed the warning but tags are still work in progress and will be available soon with other blogging features like comments, archives etc.


> Menu "Themes" button "Install Theme" - error:

This was caused by a recent change in caching code, I fixed it but theme/plugin marketplace integration is not complete it currently only displays available items but install doesn't work yet.


---

The fixes are available in the latest zip.

**Comment by kvirtu** [comment]
1. You mean Style tab in page editor? I tried changing font size for an element in px and it worked, can you provide more details on exact input? - _works. is it possible to add the ability to insert numeric values ?_
2. _New template is not implemented yet but will be available soon, it will simply allow to duplicate existing template pages or start with the "blank.html" provided by the theme._ - I think blank.html is enough, but with the ability to rename
3. Maybe Menu "Edit website" rename to "Edit template" ???
4. _All parent menu items that have submenus also link directly to a page as a shortcut, not sure what shortcut is best for tools, probably cache will be the most used_ - Let there be a cache. but add in then cache top menu to the other submenus???
5. I checked /admin/?module=content/categories&type=post and titles are showing, can you provide more details regarding this issue?
[https://postimg.cc/PNWGN7qR](url)
6. _Can you provide more details like complete error message? I thinks recursive write permission is missing for public/media folder_ - I can't, it doesn't show anymore

bugs:
1. Menu Dashboard>view live site>default site > -  need to remove "." before link to site
2. Menu Page > add news page - title "Post, Posts list , save post"
3. Theme "blog" - Many bugs ( delete or redo )
4. what purpose theme "default" ?

**Comment by givanz** [comment]
> You mean Style tab in page editor? I tried changing font size for an element in px and it worked, can you provide more details on exact input? - works. is it possible to add the ability to insert numeric values ?

By numerical values you mean decimals? you can also enter decimals using a dot like 1.4, depending on browser language configuration a comma might work instead of a dot.


> New template is not implemented yet but will be available soon, it will simply allow to duplicate existing template pages or start with the "blank.html" provided by the theme. - I think blank.html is enough, but with the ability to rename

I already made the changes, default is blank.html but any other template from the theme can be used.

> Maybe Menu "Edit website" rename to "Edit template" ???

I'm not sure what name will be better suited for the action as in theory all the sections/pages of the site can be edited but in fact is the theme files that are edited which is kind of the same.

Also apart from templates individual pages/posts/products can be edited in the editor either from admin or from admin-bar (that shows on top of all pages on frontend when admin is logged in) where both the template and the content of the post/product can be updated.

Also soon there will be the option to save the page to a new template for post/product directly from the editor in this way the post/product will use a different template then the global one allowing any customization per post/product (including different header/footer etc), currently the template for 
post/product can only be changed from post/product edit page in admin.



> All parent menu items that have submenus also link directly to a page as a shortcut, not sure what shortcut is best for tools, probably cache will be the most used - Let there be a cache. but add in then cache top menu to the other submenus???

Not sure what you mean to add cache to top, it is the first item in the submenu.


> I checked /admin/?module=content/categories&type=post and titles are showing, can you provide more details regarding this issue? https://postimg.cc/PNWGN7qR

Thanks for the screenshot, I found some non ascii characters in the categories insert sql, I removed them I think this was the cause.


> Can you provide more details like complete error message? I thinks recursive write permission is missing for public/media folder - I can't, it doesn't show anymore

It works now? or it's still not working but no error message is shown?



> Menu Dashboard>view live site>default site > - need to remove "." before link to site

This is probably a bug with the url generation for host, the site url for default site is `'host'     => '*.*.*'` subdomain, domain and tld all have wildcards, probably subdomain is missing I will check the code.

> Menu Page > add news page - title "Post, Posts list , save post"

Forgot to change on page template also, I fixed it now.

> Theme "blog" - Many bugs ( delete or redo )

The theme was made before the current theme and was not updated, currently only landing is up to date but I will also update the blog theme after blogging features are finished, 

> what purpose theme "default" 

The purpose of default theme is to use it as a fallback when the current theme is missing some templates, for example /feed/ (which are used for rss and sitemap) or /emails/ which can be excluded on purpose if the theme does not need to change them and then default ones can be used.

Currently default theme support is limited and may not fallback most of the times.

**Comment by kvirtu** [comment]
1. _By numerical values you mean decimals? you can also enter decimals using a dot like 1.4, depending on browser language configuration a comma might work instead of a dot_ - I need to be able to enter numbers, for example, aside - 25% , button - 40px.
2. upload new file in Media - not working . error: <br /> <b>Warning</b>: move_uploaded_file(/virt/homes/vpu2/htdocs/public/media//public/favicon.ico): Failed to open stream: No such file or directory in <b>/virt/homes/vpu2/htdocs/admin/controller/media/media.php</b> on line <b>69</b><br /> <br /> <b>Warning</b>: move_uploaded_file(): Unable to move &quot;/virt/tmp/phpKf549a&quot; to &quot;/virt/homes/vpu2/htdocs/public/media//public/favicon.ico&quot; in <b>/virt/homes/vpu2/htdocs/admin/controller/media/media.php</b> on line <b>69</b><br /> Error uploading file!
1 Bytes
3. Menu "Settings>Content" go to /admin/?module=**admin/users**
4. Menu "Post>categories" and "Product>categories" - names are not displayed

**Comment by givanz** [comment]
> By numerical values you mean decimals? you can also enter decimals using a dot like 1.4, depending on browser language configuration a comma might work instead of a dot - I need to be able to enter numbers, for example, aside - 25% , button - 40px.

You can change from px to % from the dropdown, please see https://imgur.com/a/4PBMFx0

> Warning: move_uploaded_file(): Unable to move "/virt/tmp/phpKf549a" to "/virt/homes/vpu2/htdocs/public/media//public/favicon.ico" in /virt/homes/vpu2/htdocs/admin/controller/media/media.php on line 69 Error uploading file! 1 Bytes

Thanks for complete error message, there was a problem with upload path when apache is configured to run from / instead of /public I fixed it.

> Menu "Settings>Content" go to /admin/?module=admin/users

I fixed the link, settings > content pages are not finished yet, they should show more settings for taxonomy configuration.

> Menu "Post>categories" and "Product>categories" - names are not displayed

This was caused by some non ascii charcaters, I removed them, you will need to reinstall from the latest zip for description to show, you only need to remove /config/db.php after updating the files for reinstall to show.

---

The fixes are available in the current zip download.

**Reply by kvirtu** [reply]
probably an old archive on the site. all the bugs remain

**Reply by givanz** [reply]
I updated the zip, for titles to show for post > categories you need to reinstall, titles were missing due to sql bug on install.

**Comment by kvirtu** [comment]
If you consider yours CMS as a full-fledged web constructor, then the functions of adding and removing pages should be there. as for the removal of mandatory pages - specify it in the documentation.
I re-downloaded the archive, but some of the bugs are still there.

1. If you select any of the pages highlighted in red, it still opens the same page - the home page
[https://ibb.co/5hxCwdh](url)
2. I can't manually specify the dimensions, for example, as in the screenshot. I can't insert my own value. without this the whole point of CMS as a constructor is lost
[https://i.postimg.cc/76CGhPLf/px.jpg](url)
3. Menu "Page>Add Page" - I'm getting into the posts.
https://i.postimg.cc/1XBQZwZ6/add-page.jpg
4. categories are not displayed
https://i.postimg.cc/26Fp4hyJ/post-categories.jpg
https://i.postimg.cc/SN1B7yQy/product-caregories.jpg

P.S. As I think, if you do not fix these bugs, further testing is meaningless.
I like your idea, I hope you will listen to me.

**Comment by givanz** [comment]
> If you consider yours CMS as a full-fledged web constructor, then the functions of adding and removing pages should be there. as for the removal of mandatory pages - specify it in the documentation.
I re-downloaded the archive, but some of the bugs are still there.

I added delete template for file manager with a confirmation dialog, for core templates not sure if I will hide the delete button or add a warning to the confirmation dialog or maybe I will add a setting to enable the button for core templates.


> If you select any of the pages highlighted in red, it still opens the same page - the home page https://ibb.co/5hxCwdh

Some of the pages were variations for homepage that had the same or similar hero, I removed them until they are finished to avoid confusion.

> I can't manually specify the dimensions, for example, as in the screenshot. I can't insert my own value. without this the whole point of CMS as a constructor is lost https://i.postimg.cc/76CGhPLf/px.jpg

From the screenshot it looks like chrome on windows, I tested on chrome windows 10 and it doesn't look like in the screenshot, can you provide more info like what browser and OS was used?
I added a maxim width for units dropdown for css input this might solve the issue.

> Menu "Page>Add Page" - I'm getting into the posts. https://i.postimg.cc/1XBQZwZ6/add-page.jpg

Fixed it.

> categories are not displayed https://i.postimg.cc/26Fp4hyJ/post-categories.jpg https://i.postimg.cc/SN1B7yQy/product-caregories.jpg

I thought that the issue is caused by non ascii characters in sql and that reinstall will fix it because I couldn't reproduce the issue on dev, xampp or shared hosting account, I will try to install mysql 5.x and see if I can reproduce the bug.

---

The zip was updated with the fixes.

**Comment by kvirtu** [comment]
**Important!!! It is necessary to close access to the file directory through the browser !!! now I move freely in the directory public/admin**
1. _I added delete template for file manager with a confirmation dialog, for core templates not sure if I will hide the delete button or add a warning to the confirmation dialog or maybe I will add a setting to enable the button for core templates_ - I think a warning button will be enough. at the moment, the "Delete" button does not work
2. _Some of the pages were variations for homepage that had the same or similar hero, I removed them until they are finished to avoid confusion_. - I think it should be corrected so that there is no confusion. it may scare away many potential testers
3. _From the screenshot it looks like chrome on windows, I tested on chrome windows 10 and it doesn't look like in the screenshot, can you provide more info like what browser and OS was used? I added a maxim width for units dropdown for css input this might solve the issue._ – windows 10 LTSC , Chrome 108.0.5359.125 , Firefox 108.0.1. **I still can't add numbers. There is no such option .** no matter px, em,rem
4. And again menu "Pages".  **If I сlicking "save page", this  page add to "post"**.  You need to completely separate  page from post . According to the concept  SEO, pages like ABOUT, CONTACTS , should be in the main folder. You already have them as /page/about , and they should be /about. **redo the pages section, please**

**Reply by givanz** [reply]
> Important!!! It is necessary to close access to the file directory through the browser !!! now I move freely in the directory public/admin

Sorry I don't understand what you mean can you provide more details? Maybe you refer to apache directory listing in this case you can create .htaccess in the public/admin folder and add `Options -Indexes` I already added this option in the latest zip.
public/ folder is for assets and doesn't have php code and as it's name says it is accessible to public, if you have acesss to server configuration it is recommended set htdocs to /public instead of / for better security not to expose php files.

> I still can't add numbers. There is no such option . no matter px, em,rem

Tested the css input on chromium on windows 10 and it works, I think this might be caused by low screen resolution that doesn't allow the input to expand, can you please provide screen resolution details so I can test?

> If I сlicking "save page", this page add to "post". 

I checked the pages and add/edit page and they no longer show post, there was a problem with add page that was fixed.

> According to the concept SEO, pages like ABOUT, CONTACTS , should be in the main folder.

The default configuration for seo links is /{slug} for posts and /page/{slug} for pages this can be changed from config/routes.php but I think this is a good default for seo practices.

**Comment by kvirtu** [comment]
Thank you for listening to me ).
1. _Tested the css input on chromium on windows 10 and it works, I think this might be caused by low screen resolution that doesn't allow the input to expand, can you please provide screen resolution details so I can test?_ - Now I can enter values, but when I try to save an error appears
[https://ibb.co/cLwQptk](url)
after pressing ok, the save button is permanently in the save state
[https://ibb.co/yPLhGV0](url)
2. old bug - categories are not displayed on "Post" and "Products"
3. _The default configuration for seo links is /{slug} for posts and /page/{slug} for pages this can be changed from config/routes.php but I think this is a good default for seo practices._ - If for a blog, maybe - /date-post/{slug} ? . If for landing Page - config: /page/{slug} to /{slug}

**Reply by givanz** [reply]
> Tested the css input on chromium on windows 10 and it works, I think this might be caused by low screen resolution that doesn't allow the input to expand, can you please provide screen resolution details so I can test? - Now I can enter values, but when I try to save an error appears

This was caused by missing `backup` folder inside theme folder, I added it to zip to avoid the error, the folder must have write permission for page revisions to be saved and avoid the error message.
I changed the code to remove the loading status if there is a save error.

> old bug - categories are not displayed on "Post" and "Products"

I though that this was caused by non ascii characters inside categories data, doesn't work even after reinstall? new install should recreate the data without non ascii text.
To reinstall is enough to delete config/db.php 

> The default configuration for seo links is /{slug} for posts and /page/{slug} for pages this can be changed from config/routes.php but I think this is a good default for seo practices. - If for a blog, maybe - /date-post/{slug} ? . If for landing Page - config: /page/{slug} to /{slug}

You can change blog posts url to `/{year}-{month}-{day}/{slug}` and pages to `/{slug}` in `config/routes.php` if you wish to have this url format.
Available date parameters `{year}` `{month}` `{day}` `{hour}` `{minute}`

```php
'/{year}-{month}-{day}/{slug}' => ['module' => 'content/post/index', 'edit'=>'?module=content/post&slug={slug}'],
'/{slug}'  	 	       => ['module' => 'content/page/index', 'edit'=>'?module=content/post&slug={slug}'],
```

Date parameters were added in the last zip update.

**Comment by kvirtu** [comment]
1. I downloaded the latest archive. reinstalled, but the categories are not displayed
2. [https://ibb.co/xG0WcLF](url)
- Button "page list" goto "post list  "
- Maybe it makes sense to make the language selection drop-down list ?
3. After editing the template and saving it, when you try to go to another page, the window pops up.
[https://ibb.co/qRqpF86](url)
But the changes are saved

**Reply by givanz** [reply]
> I downloaded the latest archive. reinstalled, but the categories are not displayed

I'm unable to reproduce this issue, I temporarily disabled filtering categories by site_id in the last update can you please check if they are showing now?
Can you also check if adding new category works? maybe this is not caused by demo data insert.

> Button "page list" goto "post list "

Thanks for the report, fixed it.

> Maybe it makes sense to make the language selection drop-down list ?

When changing the language post fields like name/slug/content are changed to enter the values for the language, it makes it easier for the user to display tabs.
 
Currently for easier testing more languages than default are enabled this will change when the localization feature will be fully implemented and test languages will no longer be necessary, until then you can delete the languages from settings > localisation > languages.

> After editing the template and saving it, when you try to go to another page, the window pops up. But the changes are saved

I'm unable to reproduce this issue, I think this might be caused by old javascript files can you please try clearing browser cache?

**Comment by kvirtu** [comment]
1. _I'm unable to reproduce this issue, I temporarily disabled filtering categories by site_id in the last update can you please check if they are showing now? Can you also check if adding new category works? maybe this is not caused by demo data insert._ - the categories are now displayed.  Categories in "post"&"Products" are not added.  Categories in "post" i can edit.  In "Products" when editing:
 
 Toys111

Warning: Undefined array key 1 in /virt/homes/vpu2/htdocs/storage/compiled-templates/admin_1_default_product_categories.html on line 458

Warning: Trying to access array offset on value of type null in /virt/homes/vpu2/htdocs/storage/compiled-templates/admin_1_default_product_categories.html on line 458

Warning: Undefined array key 2 in /virt/homes/vpu2/htdocs/storage/compiled-templates/admin_1_default_product_categories.html on line 458

Warning: Trying to access array offset on value of type null in /virt/homes/vpu2/htdocs/storage/compiled-templates/admin_1_default_product_categories.html on line 458

Warning: Undefined array key 3 in /virt/homes/vpu2/htdocs/storage/compiled-templates/admin_1_default_product_categories.html on line 458

Warning: Trying to access array offset on value of type null in /virt/homes/vpu2/htdocs/storage/compiled-templates/admin_1_default_product_categories.html on line 458
2. If add new item in "Pages>Menus>Main Menu" error:
Fatal error: Uncaught Exception: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '= ?, name = ? WHERE menu_id = ?' at line 1 UPDATE menu SETslug = "222", name = "2222" WHERE menu_id = 1 - UPDATE menu SETslug = :slug, name = :name WHERE menu_id = :menu_id in /virt/homes/vpu2/htdocs/system/db/mysqli.php:395 Stack trace: #0 /virt/homes/vpu2/htdocs/storage/model/admin/menusql.php(121): Vvveb\System\Db\Mysqli->execute('UPDATE menu SET...', Array, Array) #1 /virt/homes/vpu2/htdocs/admin/controller/content/menus.php(133): Vvveb\Sql\MenuSQL->editMenu(Array) #2 [internal function]: Vvveb\Controller\Content\Menus->menu() #3 /virt/homes/vpu2/htdocs/system/core/frontcontroller.php(191): call_user_func(Array) #4 /virt/homes/vpu2/htdocs/system/core/frontcontroller.php(256): Vvveb\System\Core\FrontController::call('Vvveb\\Controlle...', 'menu', '/virt/homes/vpu...') #5 /virt/homes/vpu2/htdocs/system/core/frontcontroller.php(320): Vvveb\System\Core\FrontController::redirect('Content/Menus', 'menu') #6 /virt/homes/vpu2/htdocs/system/core/startup.php(321): Vvveb\System\Core\FrontController::dispatch() #7 /virt/homes/vpu2/htdocs/index.php(126): Vvveb\System\Core\start() #8 /virt/homes/vpu2/htdocs/admin/index.php(34): include('/virt/homes/vpu...') #9 {main} thrown in /virt/homes/vpu2/htdocs/system/db/mysqli.php on line 395
3. If edit "Products>Manufacturers>HTC" - Page not found!
(error 404)
Controller file not found!
/virt/homes/vpu2/htdocs/admin/controller/manufacturer/manufacturer.php
4. "Products>Vendor" - go to Comments
5. Themes>install theme - error:
Fatal error: Uncaught ValueError: DOMDocument::loadXML(): Argument #1 ($source) must not be empty in /virt/homes/vpu2/htdocs/system/import/rss.php:47 Stack trace: #0 /virt/homes/vpu2/htdocs/system/import/rss.php(47): DOMDocument->loadXML('', 608578) #1 /virt/homes/vpu2/htdocs/system/extensions/themes.php(104): Vvveb\System\Import\Rss->__construct(false) #2 /virt/homes/vpu2/htdocs/admin/controller/theme/market.php(39): Vvveb\System\Extensions\Themes::getMarketList(Array) #3 [internal function]: Vvveb\Controller\Theme\Market->index() #4 /virt/homes/vpu2/htdocs/system/core/frontcontroller.php(191): call_user_func(Array) #5 /virt/homes/vpu2/htdocs/system/core/frontcontroller.php(256): Vvveb\System\Core\FrontController::call('Vvveb\\Controlle...', 'index', '/virt/homes/vpu...') #6 /virt/homes/vpu2/htdocs/system/core/frontcontroller.php(320): Vvveb\System\Core\FrontController::redirect('Theme/Market', 'index') #7 /virt/homes/vpu2/htdocs/system/core/startup.php(321): Vvveb\System\Core\FrontController::dispatch() #8 /virt/homes/vpu2/htdocs/index.php(126): Vvveb\System\Core\start() #9 /virt/homes/vpu2/htdocs/admin/index.php(34): include('/virt/homes/vpu...') #10 {main} thrown in /virt/homes/vpu2/htdocs/system/import/rss.php on line 47

**Reply by givanz** [reply]
> the categories are now displayed. Categories in "post"&"Products" are not added. Categories in "post" i can edit. In "Products" when editing:

I fixed products category warning and made add new category easier to use.

> 2. If add new item in "Pages>Menus>Main Menu" error:

I fixed saving menu name/slug.

> 3. If edit "Products>Manufacturers>HTC" - Page not found!
> 4. "Products>Vendor" - go to Comments

The pages for manufacturer and vendor were missing, I added basic support until more ecommerce features are implemented.

> 5. Themes>install theme - error:

It looks like a connection issue to the server to get theme list, I added some debug code to display error messages to see what is causing the connection issue.

**Comment by kvirtu** [comment]
1. if "add item" or edit Item in "post>categories" - error:
Fatal error: Uncaught Exception: Incorrect integer value: '' for column 'taxonomy_id' at row 1 INSERT INTO taxonomy_item ( `taxonomy_id`,`parent_id` ) VALUES ( :taxonomy_item_data ) INSERT INTO taxonomy_item ( `taxonomy_id`,`parent_id` ) VALUES ( ) Array ( [0] => ss [1] => [2] => 0 ) ss in /virt/homes/vpu2/htdocs/system/db/mysqli.php:431 Stack trace: #0 /virt/homes/vpu2/htdocs/storage/model/admin/categorysql.php(2177): Vvveb\System\Db\Mysqli->execute('INSERT INTO tax...', Array, Array) #1 /virt/homes/vpu2/htdocs/admin/controller/content/categories.php(70): Vvveb\Sql\CategorySQL->addTaxonomyItem(Array) #2 [internal function]: Vvveb\Controller\Content\Categories->add() #3 /virt/homes/vpu2/htdocs/system/core/frontcontroller.php(191): call_user_func(Array) #4 /virt/homes/vpu2/htdocs/system/core/frontcontroller.php(256): Vvveb\System\Core\FrontController::call('Vvveb\\Controlle...', 'add', '/virt/homes/vpu...') #5 /virt/homes/vpu2/htdocs/system/core/frontcontroller.php(320): Vvveb\System\Core\FrontController::redirect('Content/Categor...', 'add') #6 /virt/homes/vpu2/htdocs/system/core/startup.php(322): Vvveb\System\Core\FrontController::dispatch() #7 /virt/homes/vpu2/htdocs/index.php(126): Vvveb\System\Core\start() #8 /virt/homes/vpu2/htdocs/admin/index.php(34): include('/virt/homes/vpu...') #9 {main} thrown in /virt/homes/vpu2/htdocs/system/db/mysqli.php on line
2. if "add item" or edit Item in "Products>categories" - error:
Fatal error: Uncaught Exception: Incorrect integer value: '' for column 'taxonomy_id' at row 1 INSERT INTO taxonomy_item ( `taxonomy_id`,`parent_id` ) VALUES ( :taxonomy_item_data ) INSERT INTO taxonomy_item ( `taxonomy_id`,`parent_id` ) VALUES ( ) Array ( [0] => ss [1] => [2] => 0 ) ss in /virt/homes/vpu2/htdocs/system/db/mysqli.php:431 Stack trace: #0 /virt/homes/vpu2/htdocs/storage/model/admin/categorysql.php(2177): Vvveb\System\Db\Mysqli->execute('INSERT INTO tax...', Array, Array) #1 /virt/homes/vpu2/htdocs/admin/controller/product/categories.php(70): Vvveb\Sql\CategorySQL->addTaxonomyItem(Array) #2 [internal function]: Vvveb\Controller\Product\Categories->add() #3 /virt/homes/vpu2/htdocs/system/core/frontcontroller.php(191): call_user_func(Array) #4 /virt/homes/vpu2/htdocs/system/core/frontcontroller.php(256): Vvveb\System\Core\FrontController::call('Vvveb\\Controlle...', 'add', '/virt/homes/vpu...') #5 /virt/homes/vpu2/htdocs/system/core/frontcontroller.php(320): Vvveb\System\Core\FrontController::redirect('Product/Categor...', 'add') #6 /virt/homes/vpu2/htdocs/system/core/startup.php(322): Vvveb\System\Core\FrontController::dispatch() #7 /virt/homes/vpu2/htdocs/index.php(126): Vvveb\System\Core\start() #8 /virt/homes/vpu2/htdocs/admin/index.php(34): include('/virt/homes/vpu...') #9 {main} thrown in /virt/homes/vpu2/htdocs/system/db/mysqli.php on line
3. Themes>install theme - error: Curl error: 7 -

**Reply by givanz** [reply]
> if "add item" or edit Item in "post>categories" - error:

Sorry I missed this new bug, fixed in the last update.

> Themes>install theme - error: Curl error: 7 -

Error code 7 means `Failed to connect() to host` I changed the code to first try to use native php functions to get the content before trying curl and I also changed url from https to http  to rule out any https issues.
